src/NuGet.Core/NuGet.ProjectModel/PackageSpecReferenceDependencyProvider.cs

```diff
@@ -323,16 +325,9 @@ private IPackageSpecResolver GetPackageSpecResolver(string rootPath)
 
             if (!string.IsNullOrEmpty(rootPath))
             {
-                IPackageSpecResolver cachedResolver;
-                if (_resolverCache.TryGetValue(rootPath, out cachedResolver))
-                {
-                    specResolver = cachedResolver;
-                }
-                else
-                {
-                    specResolver = new PackageSpecResolver(rootPath);
-                    _resolverCache.Add(rootPath, specResolver);
-                }
+                specResolver = _resolverCache.GetOrAdd(
```

Comment:
The old code pattern wouldn't have allocated a delegate on a per call basis. Maybe leaving it as is is better